locust/test/subprocess_utils.py

```diff
@@ -0,0 +1,228 @@
+            cursor = self._stderr_cursor
+
+        start_time = time.time()
+        while time.time() - start_time < self.expect_timeout:
+            new_lines = buffer[cursor:]
+            for idx, line in enumerate(new_lines):
+                if is_match(to_expect, line):
+                    cursor += idx + 1
+                    return
+            time.sleep(0.05)
+
+        self.on_fail(
+            f"Did not see expected message: '{to_expect}' within {self.expect_timeout} seconds. Got {buffer[-5:]}"
+        )
+
+    # Check all output logs (stateless)
+    def _expect_any(self, to_expect, is_match: Callable[[Any, str], bool], *, stream="stderr"):
+        __tracebackhide__ = True
+
+        if stream == "stdout":
+            buffer = self.stdout_output
+        else:
+            buffer = self.stderr_output
+
+        if any(is_match(to_expect, line) for line in buffer):
+            return
+
+        self.on_fail(f"Did not see expected message: '{to_expect}'. Got {buffer[-5:]}")
+
+    def _not_expect_any(self, to_not_expect, is_match: Callable[[Any, str], bool], *, stream="stderr"):
+        __tracebackhide__ = True
+
+        if stream == "stdout":
+            buffer = self.stdout_output
+        else:
+            buffer = self.stderr_output
+
+        if any(is_match(to_not_expect, line) for line in buffer):
+            self.on_fail(f"Found unexpected message: '{to_not_expect}'.")
+
+    def expect(self, output: str, **kwargs):
+        __tracebackhide__ = True
+
+        is_match: Callable[[str, str], bool] = lambda out, line: out in line
+        return self._expect(output, is_match, **kwargs)
+
+    def expect_any(self, output: str, **kwargs):
+        __tracebackhide__ = True
+
+        is_match: Callable[[str, str], bool] = lambda out, line: out in line
+        return self._expect_any(output, is_match, **kwargs)
+
+    def not_expect_any(self, output: str, **kwargs):
+        __tracebackhide__ = True
+
+        is_match: Callable[[str, str], bool] = lambda out, line: out in line
+        return self._not_expect_any(output, is_match, **kwargs)
+
+    def expect_regex(self, pattern: str | re.Pattern[str], **kwargs):
+        __tracebackhide__ = True
+
+        if isinstance(pattern, str):
+            regex = re.compile(pattern)
+        else:
+            regex = pattern
+
+        is_match: Callable[[re.Pattern, str], bool] = lambda pattern, line: pattern.search(line) is not None
+        return self._expect(regex, is_match, **kwargs)
+
+    def expect_regex_any(self, pattern: str | re.Pattern[str], **kwargs):
+        __tracebackhide__ = True
+
+        if isinstance(pattern, str):
+            regex = re.compile(pattern)
+        else:
+            regex = pattern
+
+        is_match: Callable[[re.Pattern, str], bool] = lambda pattern, line: pattern.search(line) is not None
+        return self._expect(regex, is_match, **kwargs)
+
+    def send_input(self, content: str):
+        if self.use_pty:
```

Comment:
Prefer to crash instead of silently ignoring if someone tries to send input when use_pty is false.